Public/Src/Cache/ContentStore/Library/Utils/ResultsExtensions.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Diagnostics.ContractsLight;
+using BuildXL.Cache.ContentStore.Interfaces.Results;
+using BuildXL.Cache.ContentStore.Synchronization;
+
+namespace BuildXL.Cache.ContentStore.Utils
+{
+    internal static class ResultsExtensions
+    {
+        /// <nodoc />
+        public static TResult WithLockAcquisitionDuration<TResult, TLockKey>(this TResult result, in LockSet<TLockKey>.LockHandle handle)
+            where TResult : ResultBase
+            where TLockKey : IEquatable<TLockKey>
+        {
+            Contract.Requires(!result.Succeeded || result.Diagnostics == null, $"Diagnostics property can be set only once.");
+
+            if (result.Succeeded && handle.LockAcquisitionDuration != null)
+            {
+                var message = $", LockWait={(long)handle.LockAcquisitionDuration.Value.TotalMilliseconds}ms";

```

Comment:
>message [](start = 20, length = 7)

This local variable is unnecessary

Reply:
> > ```
> >     protected string GetErrorString()
> > ```
> 
> Will this now return `Error=[Something went wrong] Diagnostics=[, LockWait=42ms]`?
> 
> Refers to: Public/Src/Cache/ContentStore/Interfaces/Results/ResultBase.cs:157 in [419fc8c](https://github.com/Microsoft/BuildXL/commit/419fc8ce01f383432b5128b7f64819ba04d703fa). [](commit_id = [419fc8c](https://github.com/Microsoft/BuildXL/commit/419fc8ce01f383432b5128b7f64819ba04d703fa), deletion_comment = False)

It won't because you can set lock duration only for successful case.